_shared/project/tests/unit/pytest_plugins/cookiecutter.py

```diff
@@ -0,0 +1,64 @@
+import pytest
+{% if cookiecutter.get("_directory") == "pyramid-app" %}
+from pyramid import testing
+from pyramid.request import apply_request_extensions
+{% endif %}
+{% if cookiecutter.get("postgres") == "yes" %}
+
+from tests.factories.factoryboy_sqlalchemy_session import (
+    clear_factoryboy_sqlalchemy_session,
+    set_factorboy_sqlalchemy_session,
+)
+{% endif %}
+{% if cookiecutter.get("postgres") == "yes" %}
+
+
+@pytest.fixture
+def db(db_engine, db_sessionfactory):
+    """Return the SQLAlchemy database session.
+
+    This returns a session that is wrapped in an external transaction that is
+    rolled back after each test, so tests can't make database changes that
+    affect later tests.  Even if the test (or the code under test) calls
+    session.commit() this won't touch the external transaction.
+
+    This is the same technique as used in SQLAlchemy's own CI:
+    https://docs.sqlalchemy.org/en/20/orm/session_transaction.html#joining-a-session-into-an-external-transaction-such-as-for-test-suites
+    """
+    connection = db_engine.connect()
+    transaction = connection.begin()
+    session = db_sessionfactory(
+        bind=connection, join_transaction_mode="create_savepoint"
+    )
+    set_factorboy_sqlalchemy_session(session)
+
+    yield session
+
+    clear_factoryboy_sqlalchemy_session()
+    session.close()
+    transaction.rollback()
+    connection.close()
```

Comment:
This is the isolated SQLAlchemy session/transaction that's compatible with running the unittests in parallel. This PR doesn't actually get the unittests running in parallel: I'll do that in a separate PR because there's other issues to work out with it. So for now this is just a fast and SQLAlchemy-recommended way of resetting the DB between unittests that're running in serial. But it's also necessary to enable parallel unittests in future